tiledb/sm/subarray/range_subset.h

```diff
@@ -141,7 +140,78 @@ struct SortStrategy<std::string, std::string> {
           return a.start_str() < b.start_str() ||
                  (a.start_str() == b.start_str() && a.end_str() < b.end_str());
         });
-    return Status::Ok();
+  };
+};
+
+/** Default merge behavior: merging is not enabled. */
+template <typename T, typename Enable = T>
+struct MergeStrategy {
+  static void merge(std::vector<Range>&);
+};
+
+/** Merge algorithm for numeric-type ranges. */
+template <typename T>
+struct MergeStrategy<
+    T,
+    typename std::enable_if<std::is_arithmetic<T>::value, T>::type> {
+  static void merge(std::vector<Range>& ranges) {
+    auto head = ranges.begin();
+    size_t merged_cells = 0;
+
+    // Merge
+    for (auto tail = head + 1; tail != ranges.end(); tail++) {
```

Comment:
Make two new range function to simplify those conditions:

`template <typename T>`
`  inline T start_as() const {`
`    assert(!var_size_);`
`    assert(!range_.empty());`
`    return *static_cast<const T*>(range_.data());`
`  }`

`template <typename T>`
`  inline T end_as() const {`
`    assert(!var_size_);`
`    assert(!range_.empty());`
`    auto end_pos = range_.size() / 2;`
`    return &range_[end_pos];`
`  }`

I think we can also delete this function in range:
inline const T* typed_data() const.